cmd/builder/internal/builder/config.go

```diff
@@ -220,7 +221,7 @@ func validateModules(name string, mods []Module) error {
 	return nil
 }
 
-func parseModules(mods []Module) ([]Module, error) {
+func parseModules(mods []Module, usedNames map[string]int) ([]Module, map[string]int, error) {
```

Comment:
```suggestion
func parseModules(mods []Module, usedNames map[string]int) ([]Module, error) {
```

Maps are reference types so modifying it within the function is enough to achieve the desired effect for the caller. ([example](https://go.dev/play/p/dLL21ygzt3D))

Reply:
Good call, updated 👍